src/enlyze/client.py

```diff
@@ -265,3 +278,71 @@ def get_timeseries_with_resampling(
             end=end,
             variables=variables_sequence,
         )
+
+    def _get_production_runs(
+        self,
+        *,
+        production_order: Optional[str] = None,
+        product: Optional[str] = None,
+        appliance: Optional[UUID] = None,
+        start: Optional[datetime] = None,
+        end: Optional[datetime] = None,
+    ) -> Iterator[ProductionRun]:
+        """Get production runs from the API."""
+
+        filters = {
+            "production_order": production_order,
+            "product": product,
+            "appliance": appliance,
+            "start": start.isoformat() if start else None,
+            "end": end.isoformat() if end else None,
+        }
+        params = {k: v for k, v in filters.items() if v is not None}
+        return self._production_runs_api_client.get_paginated(
+            "production-runs", ProductionRun, params=params
+        )
+
+    def get_production_runs(
+        self,
+        *,
+        production_order: Optional[str] = None,
+        product: Optional[user_models.Product | str] = None,
+        appliance: Optional[user_models.Appliance] = None,
+        start: Optional[datetime] = None,
+        end: Optional[datetime] = None,
+    ) -> user_models.ProductionRuns:
+        """Retrieve optionally filtered list of :ref:`production runs <production_run>`.
+
+        :param appliance: The appliance for which to get all production runs.
+        :param product: Filter production runs by product.
+        :param production_order: Filter production runs by production order.
+
+        :raises: |token-error|
+
+        :raises: |generic-error|
+
+        :returns: Production runs
+        :rtype: :class:`~enlyze.models.ProductionRuns`
+
+        """
+        if start and end:
+            start, end = validate_start_and_end(start, end)
+        elif start:
+            start = validate_datetime(start)
+        elif end:
+            end = validate_datetime(end)
+
+        product_filter = str(product) if product else None
```

Comment:
Let's not scatter the filtering logic and be explicit here. Moreover, `__str__()` should just provide a more user-friendly representation of an object. Here, you bake in the implicit dependency on the filter by code.

```suggestion
        product_filter = product.code if isinstance(product, Product) else product
```

Reply:
Thanks for pointing that out. At second glance, this doesn't look like a good idea at all. 1268950